### PR TITLE
Task/hw 58895 add documentation to the missing classes/function in ui sdk

### PR DESCRIPTION
**Pull request by asenthil-altmtrk**

### CODE REVIEW

File: TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift [hunk 1]
```diff
@@ -21,6 +21,7 @@ import Common
 #endif
 import UIKit
 
+/// SelectTransferMethodTypeCoordinator
```
Comment (jsaini-hw) [inline TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift]: Update to - /// Coordinator class for SelectTransferMethodType

File: TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift [hunk 2]
```diff
@@ -69,7 +70,7 @@ public class SelectTransferMethodTypeCoordinator: NSObject, HyperwalletCoordinat
         self.parentController = parentController
     }
 }
-
+/// AddTransferMethodCoordinator
```
Comment (jsaini-hw) [inline TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift]: Update to - /// Coordinator class for AddTransferMethod

File: TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift [hunk 3]
```diff
@@ -113,7 +114,7 @@ public class AddTransferMethodCoordinator: NSObject, HyperwalletCoordinator {
         self.parentController = parentController
     }
 }
-
+/// ListTransferMethodsCoordinator
```
Comment (jsaini-hw) [inline TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift]: Update to - /// Coordinator class for ListTransferMethods

File: TransferMethod/Sources/SelectTransferMethodTypeController.swift [hunk 4]
```diff
@@ -88,14 +83,15 @@ final class SelectTransferMethodTypeController: UITableViewController {
 }
 
 extension SelectTransferMethodTypeController {
+    /// - Returns  transfer method count
```
Comment (jsaini-hw) [inline TransferMethod/Sources/SelectTransferMethodTypeController.swift]: remove - from comments. In all other places as well

Comment (asenthil-altmtrk) [reply TransferMethod/Sources/SelectTransferMethodTypeController.swift]: Removed extra -

File: TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift (hunk 3 above)
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift]: Updated

File: TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift (hunk 2 above)
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift]: Updated

File: TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift (hunk 1 above)
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/Coordinators/TransferMethodCoordinators.swift]: Updated

File: Receipt/Sources/ReceiptDetailController.swift [hunk 5]
```diff
@@ -97,24 +97,25 @@ final class ReceiptDetailController: UITableViewController {
         return cell
     }
 }
-
+/// The recipt detail controller
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptDetailController.swift]: Corrected

File: Receipt/Sources/ReceiptDetailController.swift [hunk 6]
```diff
@@ -20,7 +20,7 @@
 import Common
 #endif
 import HyperwalletSDK
-
+/// The class to display detail recipt
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptDetailController.swift]: Corrected

File: Common/Sources/View/ProcessingView/ProcessingView.swift [hunk 7]
```diff
@@ -70,7 +73,7 @@ public final class ProcessingView: UIView {
         parentView.addSubview(self)
         self.layer.add(fadeInAnimation(), forKey: nil)
     }
-
+   /// Convenience Initializers
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/View/ProcessingView/ProcessingView.swift]: Removed s

File: Common/Sources/View/ProcessingView/ProcessingView.swift [hunk 8]
```diff
@@ -60,7 +63,7 @@ public final class ProcessingView: UIView {
         super.init(coder: aDecoder)
         defaultInit()
     }
-
+    /// Convenience Initializers for view
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/View/ProcessingView/ProcessingView.swift]: Removed s

File: Common/Sources/View/PasteOnlyTextField.swift [hunk 9]
```diff
@@ -15,8 +15,9 @@
 // NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-
+/// To perform past action in UITextField
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/View/PasteOnlyTextField.swift]: Corrected

File: Common/Sources/InitializationDataField.swift [hunk 10]
```diff
@@ -17,18 +17,30 @@
...
 public enum InitializationDataField: String {
+    /// The 2 letter ISO 3166-1 country code.
     case country
+    /// The 3 letter ISO 4217-1 currency code.
     case currency
+    /// The profile type. Possible values - INDIVIDUAL, BUSINESS.
     case profileType
+    /// The transfer method type. Possible values - BANK_ACCOUNT, BANK_CARD.
     case transferMethodTypeCode
+    ///  Forces to refresh the cached data.
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/InitializationDataField.swift]: Removed extra space

File: Common/Sources/InitializationDataField.swift [hunk 11]
```diff
@@ -17,18 +17,30 @@
...
     case clientTransferId
+    ///  The prepaid token
     case prepaidCardToken
+    ///  The source token
     case sourceToken
+    ///  The transfer
     case transfer
+    ///  The transfer method like bank account, bank card, PayPal account, prepaid card, paper check
     case transferMethod
+    ///   Boolean value to check whether foreign exchange rate changed
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/InitializationDataField.swift]: Removed extra space

File: Common/Sources/Helper/HyperwalletUtilViews.swift [hunk 12]
```diff
@@ -128,7 +128,7 @@ public final class SpinnerView: UIView {
         activity.startAnimating()
         return activity
     }()
-
+    /// Convenience Initializers
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Helper/HyperwalletUtilViews.swift]: Removed extra s

File: Common/Sources/Generics/GenericController.swift [hunk 13]
```diff
@@ -131,15 +137,15 @@ UISearchResultsUpdating, UISearchControllerDelegate {
...
+    /// - Returns height of headerview
     override public func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         guard #available(iOS 11.0, *) else {
             return shouldDisplaySearchBar ? searchController.searchBar.frame.size.height : CGFloat.leastNormalMagnitude
         }
 
         return CGFloat.leastNormalMagnitude
     }
-
+    /// - To select the items
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Generics/GenericController.swift]: Removed extra -

File: Common/Sources/Generics/GenericController.swift [hunk 14]
```diff
@@ -131,15 +137,15 @@ UISearchResultsUpdating, UISearchControllerDelegate {
 
         return nil
     }
-
+    /// - Returns height of headerview
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Generics/GenericController.swift]: Removed extra -

File: Common/Sources/Generics/GenericController.swift [hunk 15]
```diff
@@ -116,7 +122,7 @@ UISearchResultsUpdating, UISearchControllerDelegate {
     }
 
     // MARK: - UITableViewDelegate
-
+    /// - Returns  headerview
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Generics/GenericController.swift]: Removed extra -

File: Common/Sources/Generics/GenericCell.swift [hunk 16]
```diff
@@ -18,28 +18,29 @@
 
 import UIKit
 
-/// Represents the country and currency data to be displyed on the CountryCurrencyCell
-
+/// Represents the country and currency data to be displayed on the CountryCurrencyCell
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Generics/GenericCell.swift]: Updated

File: Common/Sources/Extensions/UIViewController.swift [hunk 17]
```diff
@@ -81,6 +83,7 @@ extension UIViewController: HyperwalletFlowDelegate {
             Holder.coordinator[ObjectIdentifier(self)] = newValue
         }
     }
+    /// Initialization of data
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/UIViewController.swift]: Updated

File: Common/Sources/Extensions/Notification.swift [hunk 18]
```diff
@@ -44,14 +44,13 @@ public extension Notification.Name {
 }
 
 /// The Hyperwallet's `NotificationCenter` key to access the information.
-///
-/// - transferMethodAdded: A new transfer method has been added.
-/// - transferMethodDeactivated: A transfer method has been deactivated.
-/// - transferCreated: A transfer of funds has been created.
-/// - transferScheduled: A transfer has been scheduled.
 public enum UserInfo: String {
+    /// - transferMethodAdded: A new transfer method has been added.
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/Notification.swift]: Removed in all other places as well

File: TransferMethodRepository/Sources/TransferMethodConfigurationRepository.swift [hunk 19]
```diff
@@ -48,7 +48,7 @@ public protocol TransferMethodConfigurationRepository {
     func refreshKeys()
 }
 
-// MARK: - RemoteTransferMethodConfigurationRepository
+/// RemoteTransferMethodConfigurationRepository
```
Comment (ddhar-hw) [inline TransferMethodRepository/Sources/TransferMethodConfigurationRepository.swift]: Think `MARK` and documentation serve different purposes - please keep the `MARK`

File: UserRepository/Sources/UserRepository.swift [hunk 20]
```diff
@@ -30,7 +30,7 @@ public protocol UserRepository {
     func refreshUser()
 }
 
-// MARK: - RemoteUserRepository
```
Comment (ddhar-hw) [inline UserRepository/Sources/UserRepository.swift]: Same here

File: TransferMethod/Sources/View/Widget/TextWidget.swift [hunk 21]
```diff
@@ -111,36 +110,41 @@ class TextWidget: AbstractWidget {
...
-                    applyFormatForPatternCharacter(currentText: currentText,
+                    applyFormatForPatternCharacter(currentPatternCharacter: currentPatternCharacter,
+                                                   currentText: currentText,
+                                                   currentTextCharacter: currentTextCharacter,
+                                                   currentTextIndex: &currentTextIndex,
                                                    finalText: &finalText,
-                                                   currentIndex: &currentIndex,
+                                                   isEscapedCharacter: &isEscapedCharacter,
                                                    pattern: pattern,
+                                                   patternIndex: &patternIndex,
```
Comment (vwagh-hw) [inline TransferMethod/Sources/View/Widget/TextWidget.swift]: it looks like this PR is reverting my changes.. is this intended code change?

File: TransferMethod/Sources/View/Widget/TextWidget.swift [hunk 22]
```diff
@@ -111,36 +110,41 @@ class TextWidget: AbstractWidget {
...
                     }
 
-                    applyFormatForPatternCharacter(currentText: currentText,
+                    applyFormatForPatternCharacter(currentPatternCharacter: currentPatternCharacter,
+                                                   currentText: currentText,
+                                                   currentTextCharacter: currentTextCharacter,
+                                                   currentTextIndex: &currentTextIndex,
                                                    finalText: &finalText,
-                                                   currentIndex: &currentIndex,
+                                                   isEscapedCharacter: &isEscapedCharacter,
```
Comment (vwagh-hw) [inline TransferMethod/Sources/View/Widget/TextWidget.swift]: this parameters were removed as part of refactor and other code in this class.. something is wrong in this PR

**Pull request merged**